plugins/PrivacyManager/tests/Fixtures/RandomizedConfigIdVisitsFixture.php

```diff
@@ -0,0 +1,147 @@
+    {
+        // empty
+    }
+
+    private function setConfigIdRandomisationPrivacyConfig(bool $config)
+    {
+        $this->privacyManagerConfig->randomizeConfigId = $config;
+    }
+
+    private function addHour()
+    {
+        $this->dateTime = Date::factory($this->dateTime)->addPeriod(1, 'hour')->getDatetime();
+    }
+
+    private function setUpWebsite()
+    {
+        if (!self::siteCreated($this->idSite)) {
+            $idSite = self::createWebsite($this->dateTime, $ecommerce = 1);
+            $this->assertSame($this->idSite, $idSite);
+        }
+    }
+
+    protected function trackStandardVisits(int $visits)
+    {
+        $t = self::getTracker($this->idSite, $this->dateTime, $defaultInit = true);
+        $t->setUrl('http://example.com/');
+        for ($v = 1; $v <= $visits; $v++) {
+            $t->setForceVisitDateTime(Date::factory($this->dateTime)->addPeriod($v, 'minute')->getDatetime());
+            self::checkResponse($t->doTrackPageView("Standard visit - $v"));
+        }
+    }
+
+    protected function trackVisitsWithMultipleActions(int $visits, int $actions)
+    {
+        for ($v = 1; $v <= $visits; $v++) {
+            $t = self::getTracker($this->idSite, $this->dateTime, $defaultInit = true);
+            $t->setUrl('http://example.com/');
+            $t->setForceVisitDateTime(Date::factory($this->dateTime)->addPeriod($v, 'minute')->getDatetime());
+
+            self::checkResponse($t->doTrackPageView("Visit with actions - $v"));
+            for ($a = 1; $a <= $actions; $a++) {
+                $t->setForceVisitDateTime(
+                    Date::factory($this->dateTime)
+                        ->addPeriod($v, 'minute')
+                        ->addPeriod($a, 'second')
+                        ->getDatetime()
+                );
+                self::checkResponse($t->doTrackAction("http://example.com/$v-$a", 'link'));
+            }
+        }
+    }
+
+    protected function trackVisitsWithUserId(int $visits)
+    {
+        $t = self::getTracker($this->idSite, $this->dateTime, $defaultInit = true);
+        $t->setUserId('foobar');
+        $t->setUrl('http://example.com/');
+        for ($v = 1; $v <= $visits; $v++) {
+            $t->setForceVisitDateTime(Date::factory($this->dateTime)->addPeriod($v, 'minute')->getDatetime());
+            self::checkResponse($t->doTrackPageView("Visit with user ID set - $v"));
+        }
+    }
+
+    protected function trackEcommerceOrder(int $orders)
+    {
+        $t = self::getTracker($this->idSite, $this->dateTime, $defaultInit = true);
+        $t->setUrl('http://example.com/myorder');
+        self::checkResponse($t->doTrackPageView('Visit with ecommerce order'));
+
+        for ($o = 1; $o <= $orders; $o++) {
+            $t->setForceVisitDateTime(Date::factory($this->dateTime)->addPeriod($o, 'second')->getDatetime());
+            $t->doTrackEcommerceOrder('Ecommerce order ID - ' . rand(1, 1000000), 10 * rand(1, 100), 7, 2, 1, 0);
```

Comment:
We should avoid anything random in the fixture if possible.

Especially if the `OmniFixture` picks this fixture up and then becomes random itself.

Reply:
Fair enough. I will use the date time of the action since when it was just using the iterator from the loop they weren't unique enough for easy debugging.